src/index.ts

```diff
@@ -1,30 +1,40 @@
+export {
+  PaymentMethod,
+  PaymentAction,
+  PaymentMethodsResponse,
+  StoredPaymentMethod,
+  PaymentAmount,
+  Card,
+  PaymentResponse,
+  PaymentMethodData,
+} from './Core/types';
+
 export {
   AdyenDropIn,
   AdyenApplePay,
   AdyenInstant,
   AdyenGooglePay,
   AdyenCSE,
+  AdyenActionComponent,
 } from './AdyenNativeModules';
 
-export { AdyenCheckout, AdyenCheckoutContext } from './AdyenCheckoutContext';
-
 export {
-  PAYMENT_COMPLETED_EVENT,
-  PAYMENT_SUBMIT_EVENT,
-  PAYMENT_PROVIDE_DETAILS_EVENT,
-  PAYMENT_FAILED_EVENT,
-} from './constants';
+  AdyenCheckout,
+  AdyenCheckoutContext,
+  AdyenCheckoutProps,
+  AdyenError,
+} from './AdyenCheckoutContext';
 
-export {
-  ERROR_CODE_CANCELED,
-  ERROR_CODE_NOT_SUPPORTED,
-  ERROR_CODE_NO_CLIENT_KEY,
-  ERROR_CODE_NO_PAYMENT,
-  ERROR_CODE_INVALID_PAYMENT_METHODS,
-  ERROR_CODE_INVALID_ACTION,
-  ERROR_CODE_NO_PAYMENT_METHOD,
-} from './constants';
+export { Environment, Event, ErrorCode, ResultCode } from './Core/constants';
 
-export { Environment, Event, ErrorCode, ResultCode } from './constants';
+export {
+  Configuration,
+  DropInConfiguration,
+  CardAuthMethod,
+  ApplePayConfiguration,
+  GooglePayConfiguration,
+  AddressMode,
+  FieldVisibility,
+} from './Core/configuration';
```

Comment:
Did you try by any chance to integrate these changes within a React Native app written in Typescript?  Ex: build the library files, and then import it within other app 

That can be a good way to test if the types are working as expected . For now, your code is typed and it works fine on your IDE and when developing, but it is important to double-check if it also works well when other Typescript app start to integrate with it.

Reply:
Yes. Checked on TS Demo project 😎